### DISCUSSION TITLE
What is the difference between redirecting page A to B and move to B in page A

### DISCUSSION THREAD

**Discussion by givvemee** [pmndrs/jotai#2561] [Q&A] [answered]
I want to show code first.
My src/store/user/index.ts (atom store)
```
import { atom } from "jotai";

export const User = atom({
  email: "",
  nickname: "",
});


```
I have some codes using `useAtom` in home.tsx

```
"use client";
import { User } from "@/src/store/user";
import { useAtom } from "jotai";

const Form = ({ type }: FormType) => {
  const [user, setUser] = useAtom(User);
  // ... 
  const handleLogIn = async (data: InputType) => {
    await logIn({
      variables: {},
      },
    })
      .then(async () => {
        await setUser({
          email: data.email,
          nickname: getNickname(data.email),
        });
        handleRedirect("/pageA");
      })
      .catch((error) => {
       // ...
      });
  };

  return (
     // ... 
  );
};

export default Form;
```

When I success to login with function `handleLogIn`, I have user information `{email: 'myMail', nickname: 'myNickname" }` in "PageA". But I lost user when I move myself directly to "PageB" by changing url address like `{email: '', nickname: ''}`. Is it normallay happen? When I make a button in "PageA" that simply redirect to "PageB" with push in `useRouter()` like 
```
<button onClick={() => push("/pageB")}>test</button>
```
i do not lost my user information. 
What is the difference between two?

**Comment by dai-shi** [comment]
Hope someone can help on this.

Is this about Next.js? (I first thought it's Waku.)

**Reply by givvemee** [reply]
@dai-shi Yes, it's Next js project

**Comment by givvemee** [comment, marked as answer]
it happens becuase of non existence of persist. ..
